riscv/src/register/macros.rs

```diff
@@ -10,14 +10,14 @@ macro_rules! read_csr {
         #[inline]
         unsafe fn _read() -> usize {
             match () {
-                #[cfg(riscv)]
+                #[cfg(target_arch = "riscv64")]
```

Comment:
I think replacing `riscv` -> `target_arch = "riscv64"` in this file is not correct: https://github.com/rust-embedded/riscv/pull/203#discussion_r1590188559

Reply:
This was my mistake, I'll fix it in the next set of fixups to this PR. Along with the comments about non-`_rv32` macros.